src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -324,7 +326,29 @@ private int calculateNumberOfBatches(int listSize) {
      */
     private void handleError(@Nonnull final String errorMessage, @Nullable final Throwable exception,
                              final int failedTimes) {
-        syncOptions.applyErrorCallback(errorMessage, exception);
+        SyncException syncException = exception != null ? new SyncException(errorMessage, exception)
+            : new SyncException(errorMessage);
+        syncOptions.applyErrorCallback(syncException, null, null, null);
```

Comment:
I think I don't fully understand what you mean. 
There are two kinds of method `handleError` One is called when we don't have new resource or draft (f.e: `InventoryEntry#fetchExistingInventories`), and second when those values are known (f.e: `InventoryEntry#create`). And the second one is being called in situation you described.

Reply:
`InventorySync.java:validateDraft()` we could pass `inventoryDraft` to error callback in case SKU is null. Maybe there are more cases like this where we. Same applies for `InventorySync.java:syncBatch()`.